### ISSUE TITLE
Window content not visible in full screen mode on Cocoa

### ISSUE THREAD

**Issue by freakboy3742** [beeware/toga#2796]
_Originally posted by @proneon267 in https://github.com/beeware/toga/issues/2473#issuecomment-2317588986_

On macOS, if the content of the window is a toga.Box(), and the window is made full screen (full app-level presentation mode, not window-level full-screen), the window content is not visible.

#### If `main_window.content` is a `toga.Box`:
```python
"""
My first application
"""

import toga

class HelloWorld(toga.App):
    def handle_press(self,widget,**kwargs):
        self.set_full_screen(self.main_window)
        self.main_window.content.add(toga.Label(text="We are in App Full Screen mode"))
        print(f"Subviews After App Full Screen:{self.main_window._impl.native.contentView.subviews}")
        
    def startup(self):
        """Construct and show the Toga application.

        Usually, you would add your application to a main content box.
        We then create a main window (with a name matching the app), and
        show the main window.
        """

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = toga.Box(children=[
            toga.Button(text="Enter App Full Screen Mode",on_press=self.handle_press),
            toga.Label(
                text=""" Lorem ipsum dolor sit amet, consectetur adipiscing elit.\n"""
                     """ Maecenas vitae felis non est vehicula pulvinar id eu augue.\n"""
                     """ Maecenas et lectus tellus. Sed et lacinia erat, at pharetra quam.\n"""
                     )
        ])
        self.main_window.show()
        print(f"Subviews Before App Full Screen:{self.main_window._impl.native.contentView.subviews}")


def main():
    return HelloWorld()
```
![image](https://github.com/user-attachments/assets/05125734-b66c-40d8-9ae5-fd9dffd3c34a)

After entering app full screen mode:
![Screenshot 2024-08-29 at 8 46 36 AM](https://github.com/user-attachments/assets/ffc6fbc9-2a68-4c9f-938b-94dec8b5b0bc)

The children of the toga.Box() are not visible after entering full screen.

#### However, if the `main_window.content` is a `toga.ScrollContainer` then it works correctly:
```python
"""
My first application
"""

import toga

class HelloWorld(toga.App):
    def handle_press(self,widget,**kwargs):
        self.set_full_screen(self.main_window)
        self.main_window.content.content.add(toga.Label(text="We are in App Full Screen mode"))
        print(f"Subviews After App Full Screen:{self.main_window._impl.native.contentView.subviews}")
        
    def startup(self):
        """Construct and show the Toga application.

        Usually, you would add your application to a main content box.
        We then create a main window (with a name matching the app), and
        show the main window.
        """

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = toga.ScrollContainer(content=toga.Box(children=[
            toga.Button(text="Enter App Full Screen Mode",on_press=self.handle_press),
            toga.Label(
                text=""" Lorem ipsum dolor sit amet, consectetur adipiscing elit.\n"""
                     """ Maecenas vitae felis non est vehicula pulvinar id eu augue.\n"""
                     """ Maecenas et lectus tellus. Sed et lacinia erat, at pharetra quam.\n"""
                     )
        ]))
        self.main_window.show()
        print(f"Subviews Before App Full Screen:{self.main_window._impl.native.contentView.subviews}")


def main():
    return HelloWorld()
```
![Screenshot 2024-08-29 at 8 49 28 AM](https://github.com/user-attachments/assets/fec96cd4-10a1-470b-bd85-f5b3504377b8)

The children are being added properly as indicated by the subviews, but they are not visible when content is a `toga.Box()`

**Comment by freakboy3742** [comment]
My guess is that the problem will be a layout issue - when going full screen, the "window" is possible reporting as "zero sized" (either temporarily or permanently), resulting in an incorrect layout for window content.

**Comment by proneon267** [comment]
Thanks for shifting it here. I'll look into this and report back.

**Comment by freakboy3742** [comment]
Fixed by #2800.
